### PR TITLE
feat(insights): trends hogql volume

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

There are only a limited number of ways to display trends volume.

## Changes

Adds HogQL expressions

## How did you test this code?

- Added an E2E test
- Added django unit test

### CODE REVIEW

**Comment by mariusandra** [comment]
I couldn't get person property access working easily. Proposing we launch it without. Just how to display this? 🤔 

image

File: ee/clickhouse/queries/test/__snapshots__/test_paths.ambr
Comment (mariusandra) [inline ee/clickhouse/queries/test/__snapshots__/test_paths.ambr:1]: seems unrelated 🤔 🤷

File: frontend/src/queries/utils.ts [hunk 1]
```diff
@@ -241,19 +241,6 @@ export function taxonomicFilterToHogQl(
...
-        return [TaxonomicFilterGroupType.PersonProperties, hogQl.substring(18)]
-    }
-    if (hogQl.startsWith('properties.$feature/')) {
-        return [TaxonomicFilterGroupType.EventFeatureFlags, hogQl.substring(11)]
-    }
-    if (hogQl.startsWith('properties.')) {
-        return [TaxonomicFilterGroupType.EventProperties, hogQl.substring(11)]
-    }
-    return [TaxonomicFilterGroupType.HogQLExpression, hogQl]
-}
```
Comment (mariusandra) [inline frontend/src/queries/utils.ts:255]: This wasn't used anywhere 🤷

File: frontend/src/scenes/insights/filters/ActionFilter/ActionFilterRow/ActionFilterRow.tsx [hunk 2]
```diff
@@ -472,6 +522,18 @@ function useMathSelectorOptions({
...
         })
     }
+
+    options.push({
+        value: HogQLMathType.HogQL,
+        label: (
+            <div className="flex items-center gap-2">
+                <span>HogQL Expression</span>
+            </div>
+        ),
```
Comment (Twixes) [inline frontend/src/scenes/insights/filters/ActionFilter/ActionFilterRow/ActionFilterRow.tsx:532]:
```suggestion
       label: 'HogQL expression',
```

File: frontend/src/queries/schema.ts [hunk 3]
```diff
@@ -124,8 +125,9 @@ export interface HogQLQuery extends DataNode {
 export interface EntityNode extends DataNode {
     name?: string
     custom_name?: string
-    math?: BaseMathType | PropertyMathType | CountPerActorMathType | GroupMathType
+    math?: BaseMathType | PropertyMathType | CountPerActorMathType | GroupMathType | HogQLMathType
     math_property?: string
+    math_hogql?: string
     math_group_type_index?: 0 | 1 | 2 | 3 | 4
```
Comment (Twixes) [inline frontend/src/queries/schema.ts:131]: In terms of data modeling, this should really be just one `math` property, as it doesn't make sense for e.g `math_hogql` to be set if `math` is not `HogQLMathType` – or vice versa.
Ideally this'd be like:
`math: TrendsMath` and then `TrendsMath` would be `{ type: HogQLMathType, expression: string } | { type: PropertyMathType, property: string } | ...`
Not for this PR though I suppose.

File: frontend/src/lib/components/InsightLabel/index.tsx [hunk 4]
```diff
@@ -74,6 +75,13 @@ function MathTag({ math, mathProperty, mathGroupTypeIndex }: MathTagProps): JSX.
             </>
         )
     }
+    if (math === 'hogql') {
+        return (
+            <Tag title={String(mathHogQL)}>
+                {String(mathHogQL).length > 10 ? `${mathHogQL?.substring(0, 10)}...` : String(mathHogQL ?? '')}
```
Comment (Twixes) [inline frontend/src/lib/components/InsightLabel/index.tsx:81]: The 10 character limit is a bit frustrating, having just played around with this. It's short enough that even something extremely trivial like `sum(property.amount)` gets cut off, which makes those legends effectively useless. Can we relax this limit? Ideally relying on CSS

**Review by Twixes — COMMENTED** [review]
Looks good to me, just found one tricky UI issue with truncating the expression in legends

File: frontend/src/lib/components/InsightLabel/index.tsx (hunk 4 above)
Comment (mariusandra) [reply frontend/src/lib/components/InsightLabel/index.tsx:81]: Makes sense! How about `max-w-60 text-ellipsis overflow-hidden`?

image
